### PR TITLE
[uma-auth] add disconnect button, get budget info, get real connection data for ConnectedWallet ui

### PR DESCRIPTION
**Pull request by bsiaotickchong**
- with balance permissions

- with no balance permissions

### CODE REVIEW

**Comment by bsiaotickchong** [comment]
* **#165** <a href="https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/165?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bsiaotickchong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/uma-auth-client/src/hooks/nwc-requests/useGetBudget.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  useEffect(() => {
+    const fetchGetBudget = async (nwcRequester: NwcRequester) => {
+      setIsLoading(true);
+      try {
+        const response = await nwcRequester.getBudget();
+        setGetBudgetResponse(response);
+      } catch (error) {
+        console.error("Error fetching getBudget:", error);
+        setError(error);
```
Comment (jklein24) [inline packages/uma-auth-client/src/hooks/nwc-requests/useGetBudget.tsx]: You're getting a compiler error here (and a couple other spots).

File: packages/uma-auth-client/src/hooks/nwc-requests/useGetInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { type GetInfoResponse } from "src/Nip47Types";
+import { type NwcRequester } from "src/NwcRequester";
+import { useNwcRequester } from "../useNwcRequester";
+import { useOAuth } from "../useOAuth";
+
+export const useGetInfo = () => {
```
Comment (jklein24) [inline packages/uma-auth-client/src/hooks/nwc-requests/useGetInfo.tsx:7]: I kind of wonder if there's a way for us to make these generic to have one that works for all requests with a request and response type param. They're all pretty repetitive.

File: packages/uma-auth-client/src/steps/ConnectedWallet.tsx [hunk 3]
```diff
@@ -67,28 +58,40 @@ const getLimitFrequency = (token: TokenState | undefined) => {
...
   }
 
-  if (!currency) {
+  if (!currency || !getInfoResponse || !getBudgetResponse) {
     // TODO: Add error state
-    return <div>Error loading currency data</div>;
+    return <Container>Error loading currency data</Container>;
   }
 
+  const handleDisconnect = () => {
```
Comment (jklein24) [inline packages/uma-auth-client/src/steps/ConnectedWallet.tsx:84]: Think it's worth having some sort of confirmation on this?

File: packages/uma-auth-client/src/steps/MoreOptions.tsx [hunk 4]
```diff
@@ -12,7 +11,7 @@ export const MoreOptions = () => {
   return (
     <>
       <ModalBody>
-        <Button
+        {/* <Button
```
Comment (jklein24) [inline packages/uma-auth-client/src/steps/MoreOptions.tsx:14]: nit: Maybe add a TODO note to finish implementing this for context in the code?

**Review by jklein24 — APPROVED** [review]
LGTM, but checks are failing

File: packages/uma-auth-client/src/steps/ConnectedWallet.tsx (hunk 3 above)
Comment (bsiaotickchong) [reply packages/uma-auth-client/src/steps/ConnectedWallet.tsx:84]: Flagged to design, we have one designed for the connection management screen (which i still have to add)

File: packages/uma-auth-client/src/hooks/nwc-requests/useGetInfo.tsx (hunk 2 above)
Comment (bsiaotickchong) [reply packages/uma-auth-client/src/hooks/nwc-requests/useGetInfo.tsx:7]: Yeah that's a good point; should be doable with generics? Will look into later

**Pull request merged**